contracts/contracts/vault/VaultAdmin.sol

```diff
@@ -275,12 +293,15 @@ contract VaultAdmin is VaultStorage {
             _amounts
         );
 
-        IStrategy strategyTo = IStrategy(_strategyToAddress);
-        for (uint256 i = 0; i < _assets.length; i++) {
-            require(strategyTo.supportsAsset(_assets[i]), "Asset unsupported");
+        uint256 assetCount = _assets.length;
+        for (uint256 i = 0; i < assetCount; ++i) {
+            require(
+                IStrategy(_strategyToAddress).supportsAsset(_assets[i]),
+                "Asset unsupported"
+            );
         }
         // Tell new Strategy to deposit into protocol
-        strategyTo.depositAll();
+        IStrategy(_strategyToAddress).depositAll();
```

Comment:
Many times in this PR, we've moved from something like:

```
foos[I].bar();
```

To something like:

```
foo = foos[i];
foo.bar();
```

I think this is great.

However here we're moving the other way, and I think this makes the code harder to read. Same thing a few more times in this file with strategies.

I've love to be consistent on it. I don't think the separate variable form uses more gas?

Reply:
Nick as done these as a separate PR which would be easier to review:
https://github.com/OriginProtocol/origin-dollar/pull/1578
https://github.com/OriginProtocol/origin-dollar/pull/1613

The thing is (that I've also learned via above PRs) that if you do smth like foos[I].length in as a stopping condition in for loop, it gets evaluated every loop performing an SLOAD.

Reply:
The main reason I prefer the `IStrategy(_strategyToAddress)` style is it uses one less variable which helps avoid stack too deep errors.

The gas saving in this example is very small (5 gas). 
using `strategyTo` style
```
|  Contract                               ·  Method                              ·  Min        ·  Max        ·  Avg        ·  # calls      ·  eur (avg)  │
|  OETHVault                              ·  reallocate                          ·     246075  ·     248303  ·     247189  ·            2  ·          -  │
```

using `IStrategy(_strategyToAddress)` style
```
|  Contract                               ·  Method                              ·  Min        ·  Max        ·  Avg        ·  # calls      ·  eur (avg)  │
|  OETHVault                              ·  reallocate                          ·     246070  ·     248298  ·     247184  ·            2  ·          -  │
```

Note in this case there is no gas saving in using the `assetCount` variable as `_assets` is already in memory.
```Solidity
uint256 assetCount = _assets.length;
IStrategy strategyTo = IStrategy(_strategyToAddress);
for (uint256 i = 0; i < assetCount; ++i) {
    require(strategyTo.supportsAsset(_assets[i]), "Asset unsupported");
}
```

The following uses the same gas and one less variable in the stack.
```Solidity
IStrategy strategyTo = IStrategy(_strategyToAddress);
for (uint256 i = 0; i < _assets.length; ++i) {
    require(strategyTo.supportsAsset(_assets[i]), "Asset unsupported");
}
```

But if `_assets` was a storage variable, it would definitely be worth using the `assetCount` variable as each loop will do an SLOAD of the length.

As there is no gas different, I've gone for style consistency in signing the array length to a variable.

Reply:
FYI: I've confirmed with test contracts, that when compiled with 'via-ir' optimizations, these two contracts result in identical gas usage AND identical max stack depth during evm execution.

```soliditiy
contract Variable {
	function foo(address barAddress) external {
		IBar bar = IBar(barAddress);
		bar.bazz();
	}
}

contract Cast {
	function foo(address barAddress) external {
		IBar(barAddress).bazz();
	}
}
```

Reply:
Did your `IBar` implementation do anything?

I put this in remix using Solidity 0.8.7 with the optimizer enabled
```Solidity
// SPDX-License-Identifier: MIT
pragma solidity 0.8.7;

interface IBar {
    function bazz() external;
}


contract Bar is IBar {
    uint256 counter = 1;
    function bazz() external override {
        counter += 1;
    }
}

contract Variable {
	function foo(address barAddress) external {
		IBar bar = IBar(barAddress);
		bar.bazz();
	}
}

contract Cast {
	function foo(address barAddress) external {
		IBar(barAddress).bazz();
	}
}
```

Cast execution cost 8,315
Variable execution cost 8,328
So a 13 gas saving.

Reply:
Also, I can get one more variable using cast rather than the variable method before hitting stack too deep

```Solidity
contract Cast {
    function foo(
        address addr1,
        address addr2,
        address addr3,
        address addr4,
        address addr5,
        address addr6,
        address addr7,
        address addr8,
        address addr9,
        address addr10,
        uint256 num1,
        uint256 num2,
        uint256 num3,
        uint256 num4,
        uint256 num5,
        uint256 num6
    ) external {
        IBar(addr1).bazz();
    }
}
```

```Solidity
contract Variable {
    function foo(
        address addr1,
        address addr2,
        address addr3,
        address addr4,
        address addr5,
        address addr6,
        address addr7,
        address addr8,
        address addr9,
        address addr10,
        uint256 num1,
        uint256 num2,
        uint256 num3,
        uint256 num4,
        uint256 num5
    ) external {
        IBar bar = IBar(addr1);
        bar.bazz();
    }
}
```

Reply:
The lack of EVM stack size difference in the test case was when compiled with `via-ir` - without those optimizations turned on, it does use another stack slot.

bar() was empty.

I'm okay using the casts here.